### ISSUE TITLE
Loaded question: have you considered porting to lua, integrating fully with neovim LSP?

### ISSUE THREAD

**Issue by coxackie** [hrsh7th/vim-vsnip#156]
I know that this is probably not going to go down well, but I guess it is a legitimate inquiry. So here we go:

 Neovim has a built-in LSP implementation that is getting quite stable now. VS-code (json) style snippets are the default choice for LSP integration. Which brings the natural question: `vsnip` can become the de-facto snippet implementation for neovim, especially given its great interoperability with [completion-nvim](https://github.com/nvim-lua/completion-nvim). To do this in the most efficient (in terms of speed) way, an implementation of `vsnip` in Lua would be the optimal choice.

 Any thoughts?

**Comment by hrsh7th** [comment]
I think pure Lua snippet implementation is already exists and nvim comminity accept it with welcoming.

You can see it im here. https://github.com/norcalli/snippets.nvim

**Comment by hrsh7th** [comment]
And I don't  care about licensing.
thus anyone can port it to Lua.

Currently, vsnip is large codebase so porting is a hard work.

**Comment by coxackie** [comment]
Yeah, I saw `snippets.nvim`, but unfortunately it is not "done" or well-documented, and seems abandoned - I hope I am wrong. So, without having looked at the source code yet, if someone wanted to port this in Lua using *only* integration with `completion-nvim` and built-in nvim lsp, would it be a huge effort?

**Comment by hrsh7th** [comment]
I hope `someone` port vim-vsnip to Lua...

The vim-vsnip supports `vim` and `neovim`.
And I don't want to maintenance two snippet engine...

I understood and personally agree (because I'm neovim user) but I can't do it for the above reasons.

**Comment by coxackie** [comment]
ok, many thanks, feel free to close this, if you see fit. Or leave it open if you want it to have visibility. Your call.
